helm/hydra/.rendered.yaml

```diff
@@ -0,0 +1,153 @@
+---
+# Source: hydra/templates/service.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app.kubernetes.io/name: hydra
+    helm.sh/chart: hydra-0.0.1
+    app.kubernetes.io/managed-by: Tiller
+    app.kubernetes.io/instance: hydra
+  name: hydra
+spec:
+# Provides options for the service so chart users have the full choice
+  type: "ClusterIP"
+  ports:
+    - name: http
+      port: 80
+      protocol: TCP
+      targetPort: 4444
+  selector:
+    app.kubernetes.io/name: hydra
+    app.kubernetes.io/instance: hydra
+
+---
+# Source: hydra/templates/deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: hydra
+  labels:
+    app.kubernetes.io/name: hydra
+    helm.sh/chart: hydra-0.0.1
+    app.kubernetes.io/version: 0.0.1
+    app.kubernetes.io/instance: hydra
+    app.kubernetes.io/managed-by: Tiller
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: hydra
+      app.kubernetes.io/instance: hydra
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: hydra
+        app.kubernetes.io/instance: hydra
+    spec:
+      initContainers:
+        - name: hydra-migrate
+          image: "oryd/hydra:v1.0.0-rc.8_oryOS.10"
+          imagePullPolicy: IfNotPresent
+          command: ["hydra"]
+          args: [
+              "migrate",
+              "sql",
+              "postgres://$(DB_USERNAME):$(DB_PASSWORD)@$(DB_HOST):$(DB_PORT)/$(DB_NAME)"
```

Comment:
Is this linked to postgres? Also keep in mind that we can use `-e` to load this from the enfironment ($DSN)

Reply:
At the time of this initial commit the source Chart (private to my org) uses Hydra at rev pre-dating the `.hydra.yaml` and constrained to PostgreSQL. _It is my plan to remove this constraint within the scope of this PR._

When it comes to envvars we have to be careful to walk the line between convenience and readability. From my perspective: for a cluster administrator unfamiliar with Hydra, but still required to provide the SQL storage `Resource` for whoever is requesting Hydra to be deployed, it will be better to 
- (1) explicitly see the source of DSN and 
- (2) to have a single-glance of the environment that has a major impact on the ability of Hydra to run
One way we achieve that is my opting in to listing critical environment variables directly in `PodSpec` (`deployment.yaml`) rather than _outsourcing_ them to `hydra-env` `ConfigMap`. Perhaps having an appearance of:
```
ENV1="env1"
ENV2="env2"
...
DSN="$(ENV1) plus $(ENV2) and ..."
```
will be [better][rabbitmq-example] than setting it via `args` — it's up to us to decide.

[rabbitmq-example]: https://github.com/helm/charts/blob/7b1290efb7172de960b43a9f8bc603e4c38e2594/stable/rabbitmq/templates/statefulset.yaml#L159-L179